### PR TITLE
Adjust api-controller acceptance tests after migration to istio gateway

### PR DESCRIPTION
**Pull request by kubadz**
Confirm these statements before you submit your pull request:

- [X] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [X] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [X] I have tested my changes.
- [X] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

- Adjust api-controller acceptance tests after migration to istio gateway https://github.com/kyma-project/kyma/pull/18

### CODE REVIEW

File: tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go [hunk 1]
```diff
@@ -135,7 +136,87 @@ func TestSpec(t *testing.T) {
...
+
+			t.Log("Get...")
+
+			gotApi, err := kymaInterface.GatewayV1alpha2().Apis(namespace).Get(lastApi.Name, metav1.GetOptions{})
+
+			So(err, ShouldBeNil)
+			So(gotApi, ShouldNotBeNil)
+
+			lastApi = gotApi
+
```
Comment (piotrmsc) [inline tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go]: We don't need to create gotApi object here, you can assign it directly to global one

File: tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go [hunk 2]
```diff
@@ -135,7 +136,87 @@ func TestSpec(t *testing.T) {
...
+			So(err, ShouldBeNil)
+			So(updatedApi, ShouldNotBeNil)
+
+			lastApi = updatedApi
+
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+		})
+
+		Convey("validate updated API with disabled authentication", func() {
```
Comment (piotrmsc) [inline tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go]: not needed here, make such assertion in above scenario

File: tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go [hunk 3]
```diff
@@ -135,7 +136,87 @@ func TestSpec(t *testing.T) {
...
+
+			t.Log("Validate updated...")
+
+			checkPreconditions(lastApi, t)
+			validateApiNotSecured(httpClient, lastApi)
+		})
+
+		Convey("get updated API with disabled authentication", func() {
+
```
Comment (piotrmsc) [inline tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go]: This get is not needed

File: tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go [hunk 4]
```diff
@@ -135,7 +136,87 @@ func TestSpec(t *testing.T) {
...
+			So(gotApi, ShouldNotBeNil)
+
+			lastApi = gotApi
+
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+		})
+
+		Convey("update API with custom jwt configuration", func() {
+
+			t.Log("Update...")
```
Comment (piotrmsc) [inline tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go]: Please add validation from line 220 here

Comment (kubadz) [reply tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go]: done

File: tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go (hunk 2 above)
Comment (kubadz) [reply tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go]: done

File: tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go (hunk 1 above)
Comment (kubadz) [reply tests/api-controller-acceptance-tests/apicontroller/apicontroller_test.go]: done

**Pull request merged**